test/test.js

```diff
@@ -2,180 +2,200 @@
 var Buffer = require('safe-buffer').Buffer
 var finalhandler = require('..')
 var http = require('http')
+
+var http2
+
+try {
+  http2 = require('http2')
+} catch (_err) {
+  // Nothing
+}
+
 var utils = require('./support/utils')
 
 var assert = utils.assert
 var createError = utils.createError
-var createServer = utils.createServer
+var createHTTPServer = utils.createHTTPServer
+var createHTTP2Server = utils.createHTTP2Server
 var createSlowWriteStream = utils.createSlowWriteStream
 var rawrequest = utils.rawrequest
+var rawrequestHTTP2 = utils.rawrequestHTTP2
 var request = utils.request
 var shouldHaveStatusMessage = utils.shouldHaveStatusMessage
 var shouldNotHaveBody = utils.shouldNotHaveBody
 var shouldNotHaveHeader = utils.shouldNotHaveHeader
 
-var describeStatusMessage = !/statusMessage/.test(http.IncomingMessage.toString())
-  ? describe.skip
-  : describe
+var topDescribe = function (type, createServer) {
+  var wrapper = function wrapper (req) {
+    if (type === 'http2') {
+      return req.http2()
+    }
+
+    return req
+  }
```

Comment:
Wrapper for `http2` because of https://github.com/ladjs/supertest/blob/master/test/supertest.js#L1403

Reply:
I am wondering if we could have `wrapper` make the `request` call? Then if we just renamed it to `request` all the below code would remain unchanged. I am not going to consider this blocking, and I recognize you have put in a bunch of good work getting this here, so I hate asking for more for such nitpicks, but it would be nice to have the diff be more clear.